sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -1176,8 +1176,28 @@ case class ExternalMapToCatalyst private(
   override def dataType: MapType = MapType(
     keyConverter.dataType, valueConverter.dataType, valueContainsNull = valueConverter.nullable)
 
-  override def eval(input: InternalRow): Any =
-    throw new UnsupportedOperationException("Only code-generated evaluation is supported")
+  private lazy val keyCatalystConverter =
+    CatalystTypeConverters.createToCatalystConverter(dataType.keyType)
+  private lazy val valueCatalystConverter =
+    CatalystTypeConverters.createToCatalystConverter(dataType.valueType)
+
+  override def eval(input: InternalRow): Any = {
+    val result = child.eval(input)
+    if (result != null) {
+      val mapValue = result.asInstanceOf[Map[Any, Any]]
```

Comment:
The external map can be `java.util.Map` or `scala.collection.Map`.

Reply:
ok